File: GMS-frontend/src/app/endpoints/seminari-endpoints/seminari-edit-endpoint.ts

import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {Config} from "../../config";
import {Observable} from "rxjs";
@Injectable({providedIn: 'root'})
export class SeminariEditEndpoint implements  MyBaseEndpoint<SeminariEditRequest, number>{
  constructor(public httpClient:HttpClient) { }

  Handle(request: SeminariEditRequest): Observable<number> {
    let url=Config.adresa + `Seminar-Edit`;

    return this.httpClient.post<number>(url, request);
  }

}
export interface SeminariEditRequest {
  id: number
  tema: string
  predavac: string
  datum: string
}

File: GMS-backend/GMS/Entities/Endpoint/Seminar/Edit/SeminariEditEndpoint.cs
using GMS.Data;
using GMS.Helpers;
using GMS.Helpers.Auth;
using Microsoft.AspNetCore.Mvc;

namespace GMS.Entities.Endpoint.Seminar.Edit
{
    [Route("Seminar-Edit")]
    [MyAuthorization]

    public class SeminariEditEndpoint : MyBaseEndpoint<SeminariEditRequest, int>
    {
        private readonly ApplicationDbContext db;

        public SeminariEditEndpoint(ApplicationDbContext db)
        {
            this.db = db;
        }

        [HttpPost]
        public override async Task<int> Handle([FromBody]SeminariEditRequest request, CancellationToken cancellationToken)
        {
            Models.Seminar? seminar;
            if (request.ID == 0)
            {
                seminar = new Models.Seminar();
                db.Add(seminar);


            }
            else
            {
                seminar = db.Seminar.FirstOrDefault(s => s.ID == request.ID);
                if (seminar == null)
                    throw new Exception("pogresan ID");
            }

            seminar.Tema = request.Tema.RemoveTags();
            seminar.Predavac = request.Predavac.RemoveTags();
            seminar.Datum = request.Datum;
            

            await db.SaveChangesAsync(cancellationToken);

            return seminar.ID;
        }
    }
}
